ibm/data_source_ibm_is_security_groups.go

```diff
@@ -0,0 +1,838 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	// "github.ibm.com/ibmcloud/vpc-go-sdk/vpcv1"
+)
+
+func dataSourceIBMIsSecurityGroups() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMIsSecurityGroupsRead,
+
+		Schema: map[string]*schema.Schema{
+			"resource_group": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "resource group identifier.",
+			},
+			"vpc_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "vpc identifier.",
+			},
+			"vpc_crn": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "vpc crn",
+			},
+			"vpc_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "vpc name.",
+			},
+			"security_groups": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Collection of security groups.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"created_at": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The date and time that this security group was created.",
+						},
+						"crn": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The security group's CRN.",
+						},
+						"href": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The security group's canonical URL.",
+						},
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The unique identifier for this security group.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The user-defined name for this security group. Names must be unique within the VPC the security group resides in.",
+						},
+						"network_interfaces": &schema.Schema{
```

Comment:
are there values for network interfaces in the response as it is marked deprecated?

Reply:
+ network_interfaces = [
                  + {
                      + deleted              = []
                      + href                 = "https://us-south-stage01.iaasdev.cloud.ibm.com/v1/instances/0716_9909e216-8c6c-4923-ae22-af1d30d18be1/network_interfaces/0716-a2b1fda6-51ef-469f-908c-e3f04a73110e"
                      + id                   = "0716-a2b1fda6-51ef-469f-908c-e3f04a73110e"
                      + name                 = "primary"
                      + primary_ipv4_address = "10.240.0.5"
                      + resource_type        = "network_interface"
                    },

Getting response for this.